codegen/src/test/java/org/web3j/codegen/GenerateToStringMethodTest.java

```diff
@@ -0,0 +1,73 @@
+import org.junit.jupiter.api.Test;
+import java.util.Arrays;
+import java.util.List;
+import static org.junit.jupiter.api.Assertions.*;
+import org.web3j.protocol.core.methods.response.AbiDefinition;
+import com.squareup.javapoet.MethodSpec;
+import javax.lang.model.element.Modifier;
+import java.util.ArrayList;
+import javax.lang.model.SourceVersion;
+
+
+public class GenerateToStringMethodTest {
+
+    @Test
+    public void testGenerateToStringMethod() {
+        // Step 1: Create mock components for the struct fields
+        //AbiDefinition.NamedType component1 = new AbiDefinition.NamedType("name", "string", false);
+        //AbiDefinition.NamedType component2 = new AbiDefinition.NamedType("age", "uint256", false);
+        AbiDefinition.NamedType component1 = new AbiDefinition.NamedType("name", "string", new ArrayList<>(), "", false);
+        AbiDefinition.NamedType component2 = new AbiDefinition.NamedType("age", "string", new ArrayList<>(), "", false);
+
+        // Add components to the list
+        List<AbiDefinition.NamedType> components = Arrays.asList(component1, component2);
+
+        // Step 2: Call the generateToStringMethod() with the components
+        MethodSpec toStringMethod = generateToStringMethod(components);
+
+        // Step 3: Generate the expected toString output
+        //String expectedFormat = "\"{ name='\" + name + \"', age='\" + age + \"' }\"";
+        //String expectedMethod = "return \"{ name='\" + name + \"', age='\" + age + \"' }\";";
+        //assertEquals(expectedMethod, actualMethodBody.trim(), "Generated method does not match expected output.");
+        String actualMethodBody = toStringMethod.toString();
+        System.out.println("Generated Method:\n" + actualMethodBody);
+        System.out.flush();
+        
+        StringBuilder format = new StringBuilder("\"{ ");
+        if (!components.isEmpty()) {
+            format.setLength(format.length() - 2);
+        };
+        format.append("}\"");
+        //assertTrue(actualMethodBody.contains("return \"{ name='\" + name + \"', age='\" + age + \"' }\";"));
+
+        // Step 4: Ensure the generated method matches the expected format
+        //String actualMethodBody = toStringMethod.toString();
+        //assertTrue(actualMethodBody.contains("return \"{ name='\" + name + \"', age='\" + age + \"' }\";"));
+        //assertTrue(actualMethodBody.contains(expectedFormat), "The toString method was not generated correctly.");
+        assertTrue(actualMethodBody.contains("return \"{ name='\" + name + \"', age='\" + age + \"' }\";"),
```

Comment:
```suggestion
        assertTrue(actualMethodBody.contains("return \"{ name='\" + name + \"', age='\" + age + \"',  }\";"),
```
This will fix the current test.

However I have some points about the current approach.

1. If this is just a dreft is fine, else copy /pasting the code in this class in a private method to test it is not the right thing.
2. For the moment I will stop do any changes on Solidity wrapper generator and have a look at what we have right now in this file you can find how to generate smart contracts with stucts https://blog.web3labs.com/hyperledger-web3j-truly-decode-support-for-dynamic-solidity-structs/.
3. There are 2 ways to do it, first I would recommend to check what to string returns on a TypeReference struct and if we can update only that class.
4. 2nd check the final wrapper where exactly should be the to string written when there is a struct (in both cases TypeReference and Struct class type),
By covering this we can make a plan and start implementing.